AllReadyApp/Web-App/AllReady/Areas/Admin/Controllers/CampaignManagerInviteController.cs

```diff
@@ -0,0 +1,113 @@
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace AllReady.Areas.Admin.Controllers
+{
+    [Area("Admin")]
+    [Authorize(nameof(UserType.OrgAdmin))]
+    public class CampaignManagerInviteController : Controller
+    {
+        private readonly IMediator _mediator;
+        private readonly UserManager<ApplicationUser> _userManager;
+
+        public CampaignManagerInviteController(IMediator mediator, UserManager<ApplicationUser> userManager)
+        {
+            _mediator = mediator;
+            _userManager = userManager;
+        }
+
+        // GET: Admin/Invite/Send/5
+        [Route("[area]/[controller]/[action]/{campaignId:int}")]
+        public async Task<IActionResult> Send(int campaignId)
+        {
+            var campaign = await _mediator.SendAsync(new CampaignByCampaignIdQuery { CampaignId = campaignId });
+            if (campaign == null) return NotFound();
+
+            if (!User.IsOrganizationAdmin(campaign.ManagingOrganizationId))
+            {
+                return Unauthorized();
+            }
+
+            return View("Send", new CampaignManagerInviteViewModel
+            {
+                CampaignId = campaignId,
+                CampaignName = campaign.Name,
+            });
+        }
+
+        // POST: Admin/Invite/Send/5
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        [Route("[area]/[controller]/[action]/{campaignId:int}")]
+        public async Task<IActionResult> Send(int campaignId, CampaignManagerInviteViewModel invite)
+        {
+            if (invite == null)
+            {
+                return BadRequest();
+            }
+
+            if (ModelState.IsValid)
+            {
+                var campaign = await _mediator.SendAsync(new CampaignByCampaignIdQuery { CampaignId = campaignId });
+
+                if (campaign == null)
+                {
+                    return BadRequest();
+                }
+
+                if (!User.IsOrganizationAdmin(campaign.ManagingOrganizationId))
+                {
+                    return Unauthorized();
+                }
+
+                if (await _mediator.SendAsync(new UserHasCampaignManagerInviteQuery { CampaignId = invite.CampaignId, InviteeEmail = invite.InviteeEmailAddress }))
+                {
+                    ModelState.AddModelError("InviteeEmailAddress", $"An invite already exists for {invite.InviteeEmailAddress}.");
+                    return View("Send", invite);
+                }
+
+                var userToInvite = await _userManager.FindByEmailAsync(invite.InviteeEmailAddress);
+                if (userToInvite != null && userToInvite.ManagedCampaigns.Any(m => m.CampaignId == campaignId))
+                {
+                    ModelState.AddModelError("InviteeEmailAddress", $"{invite.InviteeEmailAddress} is allready a manager for this campaign");
+                    return View("Send", invite);
+                }
```

Comment:
I'd like to move the `UserHasCampaignManagerInviteQuery` and `userToInvite.ManagedCampaigns.Any(m => m.CampaignId == campaignId)` checks earlier in the process. Try to get them into the GET Send action method so we're not allowing the user to do something that we know wont' work. We'll let the UI disallow it, instead of letting then click on stuff and then at the very end tell them, "oh, too bad, you can't perform this operation"